pkg/provision/storage/cleanup.go

```diff
@@ -203,3 +225,44 @@ func commonPVCExists(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) (bo
 	}
 	return true, nil
 }
+
+func checkCleanupPodsState(job *batchv1.Job, workspaceID string, clusterAPI sync.ClusterAPI) (msg string, err error) {
+	pods, err := check.GetPods(job.Namespace, k8sclient.MatchingLabels{"job-name": common.PVCCleanupJobName(workspaceID)}, clusterAPI.Client)
+	if err != nil {
+		return "", err
+	}
+
+	for _, pod := range pods.Items {
+
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if check.CheckContainerStatusForFailure(&containerStatus) {
```

Comment:
Currently, it's possible for the container status to be set to `waiting` with reason `ContainerCreating` which doesn't seem like it should return an error, though it will cause my patch to return a `ProvisionError`. This should be fixed IMO, though I'm not sure if the fix should be specific to the cleanup job or part of `check.CheckContainerStatusForFailure` (which will impact workspace deployments).